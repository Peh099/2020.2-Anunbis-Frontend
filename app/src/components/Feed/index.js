import React from 'react';
import {
  Container,
  DivContent,
  DivTitle,
  Grid,
  HeaderStyle,
  OrderDiv,
} from './styles';
import Post from '../Post';
import Button from '../Button';
import Popup from '../Popup';
import Avaliation from '../Avaliation';
import { orders } from '../../services/Orders';
import Select from '../Select';

const Title = (props) => {
  const { children } = props;
  return <DivTitle {...props}>{children}</DivTitle>;
};

const Header = ({
  professor,
  feedbacks,
  canAvaliate,
  onNewAvaliation,
  backColor,
  border,
}) => {
  const [avaliate, setAvaliate] = React.useState(false);

  function onClose() {
    setAvaliate(false);
    onNewAvaliation();
  }

  return (
    <>
      {avaliate && (
        <Popup>
          <Avaliation professor={professor} close={onClose} />
        </Popup>
      )}
      <HeaderStyle backColor={backColor} border={border}>
        <Grid>
          <label>Nota Geral: </label>
          <label>
            {feedbacks?.rating ? feedbacks.rating.toFixed(2) : 'Não avaliado'}
          </label>
          <label>Didática: </label>
          <label>
            {feedbacks?.didactic ? feedbacks.didactic.toFixed(2) : '0'}
          </label>
          <label>Metodologia: </label>
          <label>{feedbacks?.metod ? feedbacks.metod.toFixed(2) : '0'}</label>
          <label>Avaliações Coerentes: </label>
          <label>
            {feedbacks?.avaliations ? feedbacks.avaliations.toFixed(2) : '0'}
          </label>
          <label>Disponibilidade: </label>
          <label>
            {feedbacks?.disponibility
              ? feedbacks.disponibility.toFixed(2)
              : '0'}
          </label>
        </Grid>
        {canAvaliate && (
          <Button
            text="Avaliar"
            backColor="var(--yellow)"
            onClick={() => setAvaliate(true)}
          />
        )}
      </HeaderStyle>
    </>
  );
};

function PostsBox({ posts, canReport }) {
  const [order, setOrder] = React.useState(0);
  const orderedPost = posts.sort(orders[order].fun);
  return (
    <DivContent>
      {posts.length > 0 && (
        <OrderDiv>
          <Select
            id="orders"
            backColor="var(--transparent)"
            text="Selecione um Ordenação"
            options={orders}
            onChange={(e) => setOrder(e.target.value)}
            width="fit-content"
            display="flex"
          />
        </OrderDiv>
      )}
      {orderedPost.map((post) => (
        <Post key={post.id_post} post={post} canReport={canReport} />
      ))}
    </DivContent>
  );
}

function Feed(props) {
  const { title, children } = props;
  return (
    <Container {...props}>
      <Title>{title}</Title>
      {children}
    </Container>
  );
}

Feed.PostsBox = PostsBox;
Feed.Header = Header;
Feed.Title = Title;
export default Feed;
