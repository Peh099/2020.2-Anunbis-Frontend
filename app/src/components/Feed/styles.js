import styled from 'styled-components';

export const Container = styled.div`
  width: 450px;
  height: 450px;
  border-bottom-right-radius: 2%;
  border-bottom-left-radius: 2%;
  background-color: #FFFDE7;
  justify-content: center;
`

export const DivContent = styled.div`
    height: 400px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-around;
    overflow-y: auto;
`;

export const DivTitle = styled.div`
    width: inherit;
    height: 25px;
    background-color: #212121;
    color: #FFFDE7;
    display: flex;
    align-items: center;
    justify-content: center;
  `;