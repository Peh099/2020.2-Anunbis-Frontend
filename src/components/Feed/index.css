.Feed {
    width: 100%;
    height: 100%;
    border-bottom-right-radius: 2%;
    border-bottom-left-radius: 2%;
    background-color: #FFFDE7;
    
}

.Feed .Title {
    width: 100%;
    height: 5%;
    background-color: #212121;
    color: #FFFDE7;
    display: flex;
    align-items: center;
    justify-content: center;
}

.Feed .Content {
    width: 100%;
    height: 95%;
    display: flex;
    align-items: center;
    justify-content: center;
}