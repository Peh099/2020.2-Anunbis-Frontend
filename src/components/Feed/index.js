import React from 'react';
import "./index.css";

const Title = ({ children }) => {
  return (
    <div className="Title">
      {children}
    </div>
  )
}

export default function Feed({ title, children }) {
  return (
    <div className="Feed">
      <Title>
        {title}
      </Title>
      <div className="Content">
        {children}
      </div>
    </div>
  );
}
