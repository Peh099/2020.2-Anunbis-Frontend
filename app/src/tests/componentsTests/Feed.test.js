import renderer from 'react-test-renderer';
import Feed from '../../components/Feed';

describe('Snapshot Feed component', () => {
  it('matches the snapshot', () => {
    const tree = renderer.create(<Feed />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('matches the snapshot', () => {
    const tree = renderer.create(<Feed.PostsBox />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('matches the snapshot', () => {
    const tree = renderer.create(<Feed.Title />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('matches the snapshot', () => {
    const tree = renderer.create(<Feed.Header />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
