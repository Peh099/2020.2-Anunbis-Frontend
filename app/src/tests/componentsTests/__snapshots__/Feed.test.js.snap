// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot Feed component matches the snapshot 1`] = `
<div
  className="sc-bdnxRM dHvxIB"
>
  <div
    className="sc-gtsrHT iUwlZJ"
  />
</div>
`;

exports[`Snapshot Feed component matches the snapshot 2`] = `
<div
  className="sc-eCApnc ekjowF"
/>
`;

exports[`Snapshot Feed component matches the snapshot 3`] = `
<div
  className="sc-gtsrHT iUwlZJ"
/>
`;

exports[`Snapshot Feed component matches the snapshot 4`] = `
<div
  className="sc-jSFjdj cGLgru"
>
  <div>
    Nenhum Professor Encontrado!
  </div>
</div>
`;
